src/extractor/scripting_environment_lua.cpp

```diff
@@ -459,14 +452,127 @@ void Sol2ScriptingEnvironment::InitContext(LuaScriptingContext &context)
                               " are supported." + SOURCE_REF);
     }
 
-    // Assert that version-dependent properties were not changed by profile
+    util::Log() << "Using profile api version " << context.api_version;
+
+    // version-dependent parts of the api
     switch (context.api_version)
     {
+    case 2:
+    {
+        // clear global not used in v2
+        context.state["properties"] = sol::nullopt;
+
+        // check function table
+        if (function_table == sol::nullopt)
+            throw util::exception("Profile must return a function table.");
+
+        // call initialize function
+        sol::function setup_function = function_table.value()["setup"];
+        if (!setup_function.valid())
+            throw util::exception("Profile must have an setup() function.");
+        sol::optional<sol::table> profile_table = setup_function();
+        if (profile_table == sol::nullopt)
+            throw util::exception("Profile setup() must return a table.");
+        else
+            context.profile_table = profile_table.value();
+
+        // store functions
+        context.turn_function = function_table.value()["process_turn"];
+        context.node_function = function_table.value()["process_node"];
+        context.way_function = function_table.value()["process_way"];
+        context.segment_function = function_table.value()["process_segment"];
+        context.source_function = context.state["source_function"];
```

Comment:
actually, we still need the source_function for v1 profiles. seems we're missing a test for source_function in v1 profiles.